printThis.js

```diff
@@ -48,23 +48,39 @@
 
     function appendBody($body, $element, opt) {
         // Clone for safety and convenience
-        var $content = $element.clone();
+        // Calls clone(withDataAndEvents = true) to copy form values.
+        var $content = $element.clone(opt.formValues);
+
+        if (opt.formValues) {
+            // Copy original select and textarea values to their cloned counterpart
+            // Makes up for inability to clone select and textarea values with clone(true)
+            copyValues($element, $content, 'select, textarea');
+        }
 
         if (opt.removeScripts) {
             $content.find('script').remove();
         }
 
         if (opt.printContainer) {
             // grab $.selector as container
-            $body.append($("<div/>").html($content).html());
+            $content.appendTo($body);
         } else {
             // otherwise just print interior elements of container
             $content.each(function() {
-                $body.append($(this).html());
+                $(this).children().appendTo($body)
             });
         }
     }
 
+    // Copies values from origin to clone for passed in elementSelector
+    function copyValues(origin, clone, elementSelector) {
+        var $originalElements = origin.find(elementSelector);
+
+        clone.find(elementSelector).each(function(index, item) {
+            item.value = $originalElements.eq(index).val();
```

Comment:
Something nags me about index matching, here. I can't currently imagine a scenario in which it is problematic, though.

Reply:
`item.value` does not work as anticipated for multi-select `<select>` boxes. You need to continue to use the jQuery `.val()` for this behavior.

This change seems to do the trick:

```
$(item).val($originalElements.eq(index).val());
```